common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivityFactory.java

```diff
@@ -77,7 +79,35 @@ public static Intent getAuthorizationActivityIntent(final Context context,
                                                         final AuthorizationAgent authorizationAgent,
                                                         final boolean webViewZoomEnabled,
                                                         final boolean webViewZoomControlsEnabled) {
-        Intent intent;
+        return getAuthorizationActivityIntent(context, authIntent, requestUrl, redirectUri, requestHeaders, authorizationAgent, webViewZoomEnabled, webViewZoomControlsEnabled, null, null);
+    }
+
+    /**
+     * Return the correct authorization activity based on library configuration.
+     *
+     * @param context                    Android application context
+     * @param authIntent                 Android intent used by the authorization activity to launch the specific implementation of authorization (BROWSER, EMBEDDED)
+     * @param requestUrl                 The authorization request in URL format
+     * @param redirectUri                The expected redirect URI associated with the authorization request
+     * @param requestHeaders             Additional HTTP headers included with the authorization request
+     * @param authorizationAgent         The means by which authorization should be performed (EMBEDDED, WEBVIEW) NOTE: This should move to library configuration
+     * @param webViewZoomEnabled         This parameter is specific to embedded and controls whether webview zoom is enabled... NOTE: Needs refactoring
+     * @param webViewZoomControlsEnabled This parameter is specific to embedded and controls whether webview zoom controls are enabled... NOTE: Needs refactoring
+     * @param product                    Product name to be of library making the request
```

Comment:
(My understanding is that if this is invoked MSAL, then we'll send MSAL info. 
if it's invoked in broker, we'll send broker info. correct?)

Reply:
If it's the former - we upload the "origin" of the request, 
then such origin should be part of the Request object that we construct in the IPC layer. We should not need to use DiagnosticContext

if it's the latter - we upload the info of whoever makes the request,
then we could use the static "unmodifiable" model like the Device class.

Reply:
(regardless, we could probably set this value when we construct TokenCommandParameters in Msal/AdalAndroidBrokerCommandParameters)

Reply:
I've discussed with @shahzaibj , we agreed that using Diagnostic Context for now is fine to get the change in and keep track with other teams (this is a multi team effort, most teams are aiming to get it in this month), propagating throughout the entire code base is going to balloon these PRs quite a bit, especially since the fields already exist in diagnostic context for other purposes. Not to mention it leaves more room for misses if we start editing every flow, while Diagnostic context gets set at the command dispatcher level, so it seems like it has good coverage for our flows. We aren't adding any additional fields in Diagnostic Context, they are already there.

To answer your question though, it would be "OneAuth", not Broker. We're interested in the source library.

Reply:
We wanted to avoid a massive refactoring effort, we can discuss during scrum

Reply:
changed these to sourceLibraryName and sourceLibraryVersion for clarity

Reply:
ack. please add comment in the the library helper class (or DiagnosticContext) that we should move these two values to the parameter class at some point - and we should NEVER add anything new here unless we **_really_** need it for logging purpose.